### ISSUE TITLE
[BUG]: Wrong return type hint for Phalcon\Translate\InterpolatorFactory::newInstance

### ISSUE THREAD

**Issue by sergeyklay** [phalcon/cphalcon#14996]
`Phalcon\Translate\InterpolatorFactory::newInstance` has incorrect return type hint:
https://github.com/phalcon/cphalcon/blob/24857a7b2291b13ef41974c691fe23dea7c0071d/phalcon/Translate/InterpolatorFactory.zep#L39-L48

But this method is meant to returns an `Phalcon\Translate\Interpolator\InterpolatorInterface` instance. This leads to a fatal error:
```
PHP Fatal error:
Return value of Phalcon\Translate\InterpolatorFactory::newInstance()
must implement interface Phalcon\Translate\Adapter\AdapterInterface,
instance of Phalcon\Translate\Interpolator\AssociativeArray returned in Unknown on line 0
```

**Comment by sergeyklay** [comment]
All tests are passed. The failed one is not related (canceled build). The same as in https://github.com/phalcon/cphalcon/pull/14981

**Comment by sergeyklay** [comment]
Fixed in 4.0.x branch
